### ISSUE TITLE
coreos-metadata no longer accepts single-dash command line flags

### ISSUE THREAD

**Issue by bison** [coreos/bugs#2240]
# Issue Report #

### Container Linux Version ###

```
$ cat /etc/os-release 
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1590.0.0
VERSION_ID=1590.0.0
BUILD_ID=2017-11-08-0831
PRETTY_NAME="Container Linux by CoreOS 1590.0.0 (Ladybug)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```

### Environment ###

Bare metal on Packet.

### Expected Behavior ###

The coreos-metadata service starts with original systemd unit.

### Actual Behavior ###

During a recent alpha update, the coreos-metadata binary stopped accepting the golang style single dash command line arguments. When my hosts where deployed, they required a custom coreos-metadata systemd unit configured with Ignition, and that unit used the single dash arguments. The metadata service now fails to start, and will until either I update the Ignition and roll each host in the cluster or the binary accepts the old style flags.

**Comment by sdemos** [comment]
This is a good catch. `coreos-metadata` should continue to support the golang style single hyphen flags, instead of breaking the cli with no warning. I've added a modification that fixes this in coreos/coreos-metadata#70. It's a little bit of a hack, but since we only accept long arguments anyway, it works fine.

**Comment by bison** [comment]
Thanks for taking care of this so quickly!!!

**Comment by bgilbert** [comment]
Reopening until merged to overlay.

**Comment by bgilbert** [comment]
This should be fixed in alpha 1590.1.0 and beta 1576.3.0, due shortly.  Thanks for reporting.
